.github/actions/gas-compare/action.yml

```diff
@@ -43,7 +43,7 @@ runs:
       shell: bash
     - name: Save report
       if: github.event_name != 'pull_request'
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
-        name: gasreport
+        name: gasreport-${{ github.run_id }}-${{ github.run_attempt }}
```

Comment:
That will be an issue. 

The artefact that is uploaded here will later be downloaded by the "Download reference report" job in the same action. This step has no way to know the run_id and run_attempt that created the report.

Reply:
Hmm, yes you're right. I think this is breaking the workflow and the package owners don't seem to provide an easy workaround.

Shall we keep @v3 insetad?